### DISCUSSION TITLE
Pore model

### DISCUSSION THREAD

**Discussion by foojan-Ka** [GeoChemFoam/GeoChemFoam#12] [Q&A] [answered]
Hello,
I have a question regarding inputting my pore model to Test Case 02.
Firstly, I created my pore model and stacked the images and created 10 slices of it in image J, then after running createMesh and runSnappyMesh, in Paraview, the pores and grains are not shown properly. My pore model is displayed as a rectangular shape in Paraview without any pores and grains. My stacked image is defined in the attached picture.
![16915565589572415474189309095788](https://github.com/GeoChemFoam/GeoChemFoam/assets/88163289/eee187a2-d296-41c2-9d17-e44238175d01)
The output of Paraview is shown in the following image.
![16915571808785211852890596982116](https://github.com/GeoChemFoam/GeoChemFoam/assets/88163289/e53a3002-9506-4210-9942-5fe1b469dda5)
Could you please help me out in this?
Looking forward to hearing from you.

Best,
Foojan

**Comment by GeoChemFoam** [comment, marked as answer]
Hi Foojan,

If your pores are not connected by voxel faces (not just by edges or corners) then they will not be included in the final mesh. Are you meshing the full resolution with n_level 0?




Best wishes,

Hannah


From: Foojan-Ka ***@***.***>
Sent: Wednesday, August 9, 2023 6:02 AM
To: GeoChemFoam/GeoChemFoam ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [GeoChemFoam/GeoChemFoam] Pore model (Discussion #12)

****************************************************************
Caution: This email originated from a sender outside Heriot-Watt University.
Do not follow links or open attachments if you doubt the authenticity of the sender or the content.
****************************************************************


Hello,
I have a question regarding inputting my pore model to Test Case 02.
Firstly, I created my pore model and stacked the images and created 10 slices of it in image J, then after running createMesh and runSnappyMesh, in Paraview, the pores and grains are not shown properly. My pore model is displayed as a rectangular shape in Paraview without any pores and grains. My stacked image is defined in the attached picture.
[Image removed by sender. 16915565589572415474189309095788]<https://user-images.githubusercontent.com/88163289/259299329-eee187a2-d296-41c2-9d17-e44238175d01.jpg>
The output of Paraview is shown in the following image.
[Image removed by sender. 16915571808785211852890596982116]<https://user-images.githubusercontent.com/88163289/259300915-e53a3002-9506-4210-9942-5fe1b469dda5.jpg>
Could you please help me out in this?
Looking forward to hearing from you.

Best,
Foojan

—
Reply to this email directly, view it on GitHub<https://github.com/GeoChemFoam/GeoChemFoam/discussions/12>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AWDOAFQDEFJCW3UAPEDL5PTXUMKS5ANCNFSM6AAAAAA3JNSK3Y>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

________________________________

Founded in 1821, Heriot-Watt is a leader in ideas and solutions. With campuses and students across the entire globe we span the world, delivering innovation and educational excellence in business, engineering, design and the physical, social and life sciences. This email is generated from the Heriot-Watt University Group, which includes:

  1.  Heriot-Watt University, a Scottish charity registered under number SC000278
  2.  Heriot- Watt Services Limited (Oriam), Scotland's national performance centre for sport. Heriot-Watt Services Limited is a private limited company registered is Scotland with registered number SC271030 and registered office at Research & Enterprise Services Heriot-Watt University, Riccarton, Edinburgh, EH14 4AS.

The contents (including any attachments) are confidential. If you are not the intended recipient of this e-mail, any disclosure, copying, distribution or use of its contents is strictly prohibited, and you should please notify the sender immediately and then delete it (including any attachments) from your system.

**Reply by foojan-Ka** [reply]
Hi Hannah,

Many thanks for your reply. 
You helped me to solve the problem.

Best,
Foojan
